test/decoders/test_video_decoder_ops.py

```diff
@@ -209,19 +333,24 @@ def test_get_frames_by_pts(self, device):
         with pytest.raises(AssertionError):
             assert_frames_equal(frames[0], frames[-1])
 
+    @pytest.mark.parametrize("test_ref", (NASA_VIDEO, NASA_AUDIO))
     @pytest.mark.parametrize("device", cpu_and_cuda())
-    def test_pts_apis_against_index_ref(self, device):
+    def test_pts_apis_against_index_ref(self, test_ref, device):
         # Non-regression test for https://github.com/pytorch/torchcodec/pull/287
         # Get all frames in the video, then query all frames with all time-based
         # APIs exactly where those frames are supposed to start. We assert that
         # we get the expected frame.
-        decoder = create_from_file(str(NASA_VIDEO.path))
-        add_video_stream(decoder, device=device)
+        if device == "cuda" and test_ref is NASA_AUDIO:
+            pytest.skip(reason="CUDA decoding not supported for audio")
+        decoder = create_from_file(str(test_ref.path))
+        _add_stream(decoder=decoder, test_ref=test_ref, device=device)
 
         metadata = get_json_metadata(decoder)
         metadata_dict = json.loads(metadata)
         num_frames = metadata_dict["numFrames"]
         assert num_frames == 390
+        if test_ref is NASA_AUDIO:
+            num_frames = 204
```

Comment:
I'm confused here - does the assertion on line 351 pass for audio? If yes, why do we need to do a fixup after? I think this does call out that we'll need to reconcile our current metadata with audio.

Reply:
The assertion passes, yes. This PR doesn't address the audio metadata so all we get is wrong / related to videos.

We need to do this fixup because the audio stream only contains 204 frames, not 390. We would be trying to decode frames beyond 204 in the test, leading to a failure.